larq/metrics.py

```diff
@@ -172,3 +172,51 @@ def get_config(self):
             "values_shape": self.values_shape,
             "values_dtype": self.values_dtype.name,
         }
+
+
+@utils.register_alias("gradient_flow")
+@utils.register_keras_custom_object
+class GradientFlow(LarqMetric):
```

Comment:
@lgeiger do you know if keras supports calling a metric on gradient values? I can't quite figure it out from the [add_metric function](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/engine/base_layer.py#L1226), but I assume that calling add_metric in the backward pass will break tensorflow...

Reply:
@koenhelwegen I agree, that might be a better API, though I am also not sure how to implement this in Keras without using a custom training loop.

@jneeven I don't know. This might be good to ask in the TF/Keras mailing list.

In TensorFlow 1.x it was possible to get the gradients in the [TensorBoard callback](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/callbacks_v1.py#L211-L223). In TensorFlow 2.x [this options is disabled](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/callbacks.py#L1478-L1480).